test/hotspot/jtreg/compiler/vectorapi/VectorCommutativeOperSharingTest.java

```diff
@@ -406,4 +406,64 @@ public void testVectorIRSharingDriver18() {
             testVectorIRSharing18(i);
         }
     }
+
+    @Test
+    @IR(counts = {IRNode.ADD_VI, " 2 "})
+    public void testVectorIRSharing19(int index) {
+        VectorMask<Integer> vmask = VectorMask.fromLong(I_SPECIES, 15);
+        IntVector vec1 = IntVector.fromArray(I_SPECIES, ia, index);
+        IntVector vec2 = IntVector.fromArray(I_SPECIES, ib, index);
+        // predicated (vec1 + vec2) + (vec2 + vec1)
+        vec1.lanewise(VectorOperators.ADD, vec2, vmask)
+            .lanewise(VectorOperators.ADD, vec2.lanewise(VectorOperators.ADD, vec1, vmask))
```

Comment:
Thanks for your updating @jatin-bhateja!  

But I don't think it's right here. Do you mean `vec1.lanewise(VectorOperators.ADD, vec2, vmake)` equals to `vec2.lanewise(VectorOperators.ADD, vec1, vmask)` ?  The inactive lane values are not equal, right?  Should it keep the lane value of first input for inactive lanes? Thanks!

Reply:
Seems the predicated IR can be commutative only if the mask is `alltrue()`. I noticed the mask in this test is `alltrue`, so it maybe correct for this case. Could you please also add other IR tests that the mask is not `alltrue`? Thanks!